chromium_src/chrome/browser/lifetime/application_lifetime.cc

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "build/build_config.h"
+
+#if BUILDFLAG(IS_MAC)
+#include "brave/browser/lifetime/brave_application_lifetime_mac.h"
+
+#define AttemptRestart AttemptRestart_ChromiumImpl
+#endif
+
+#include "src/chrome/browser/lifetime/application_lifetime.cc"
+
+#if BUILDFLAG(IS_MAC)
+#undef AttemptRestart
+
+namespace chrome {
+
+void AttemptRestart() {
```

Comment:
What about `RelaunchIgnoreUnloadHandlers`?

Reply:
I wonder why not add the code only once in `AttemptRestartInternal`. I also wonder whether the code in `AttemptRestartInternal` is necessary before the Sparkle relaunch. For example, this line:
```cpp
  // Set the flag to restore state after the restart.
  pref_service->SetBoolean(prefs::kRestartLastSessionOnShutdown, true);
```
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/lifetime/application_lifetime_desktop.cc;l=124?q=AttemptRestartInternal&ss=chromium%2Fchromium%2Fsrc